tflib/publisher/main.tf

```diff
@@ -24,12 +24,33 @@ variable "extra_packages" {
   default = ["wolfi-baselayout"]
 }
 
+variable "name" {
+  type    = string
+  default = "TODO" // TODO remove this
+}
+
+output "path" {
+  value = basename(path.cwd)
+}
+
+locals {
+  updated_config = merge(yamldecode(var.config),
+    var.name == "TODO" ? {} :
+    { "annotations" = {
+      "org.opencontainers.image.authors" : "Chainguard Team https://www.chainguard.dev/",
+      "org.opencontainers.image.url" : "https://edu.chainguard.dev/chainguard/chainguard-images/reference/${var.name}/",
+      "org.opencontainers.image.source" : "https://github.com/chainguard-images/images/tree/main/images/${var.name}",
+      },
+    },
+  )
+}
+
 module "this" {
   source  = "chainguard-dev/apko/publisher"
   version = "0.0.6"
 
   target_repository = var.target_repository
-  config            = var.config
+  config            = yamlencode(local.updated_config)
```

Comment:
this surgery is gross, can we cut a release of the module with your change, bump the version here and avoid the roundtrip?

Reply:
We can, sure. I kinda thought it was nice not to do this surgery in the provider, where we need releases to change the logic, but I get that moving more stuff into HCL is not necessarily a win.

Reply:
I think for annotations the provider already does this surgery, right?

Reply:
Given the size of the change, I'd also be ok with checkpointing this and cleaning it up with the release and dropping the TODOs in follow-up.